### PR TITLE
652: Introduce --format Flag for lint Subcommand

### LINKED ISSUE #652: Add --format flag to Zally CLI
In order to make Zally Command Line Utility (CLI) usable in different Continuous Integration platforms, we want to add more formatting options. Therefore, we need to implement an extra `--format` flag. 

## Acceptance Criteria
```
As a Zally CLI user
I want to provide the --format flag when running CLI
So that Zally will format its output in plain text or markdown
```

## Example
```
./zally --format=markdown
```

## Expected text output:

```
MUST Use Problem JSON
  Operations Should Return Problem JSON When Any Problem Occurs During Processing Whether Caused by Client Or Server
  https://zalando.github.io/restful-api-guidelines/#176
    /products GET 429
    /products/{product_id} GET 418
    /products/{product_id} GET 429
    /products/{product_id} PATCH 400
    /products/{product_id} PATCH 429
```

## Expected markdown output

```markdown
### `MUST` [Use Problem JSON](https://zalando.github.io/restful-api-guidelines/#176)

> Operations Should Return Problem JSON When Any Problem Occurs During Processing Whether Caused by Client Or Server
        
- /products GET 429
- /products/{product_id} GET 418
- /products/{product_id} GET 429
- /products/{product_id} PATCH 400
- /products/{product_id} PATCH 429
```

## Expected output when no violations happened:

```
Congratulations! No violations found.
```

### PR DESCRIPTION
**Pull request by netme**
Resolves #652 

### How to test:

1. Start server locally
1. `cd cli/zally`
1. `go build .`
1. Run the following commands:
    ```bash
    ./zally lint --format markdown  ../../server/src/test/resources/fixtures/api_spa.yaml
    ./zally lint --format pretty  ../../server/src/test/resources/fixtures/api_spa.yaml 
    ./zally lint  ../../server/src/test/resources/fixtures/api_spa.yaml 
    ```

### CODE REVIEW

File: cli/zally/utils/markdown_formatter.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/zalando/zally/cli/zally/domain"
+)
+
+// MarkdownFormatter is used to generate output in pretty format
```
Comment (roxspring) [inline cli/zally/utils/markdown_formatter.go]: in markdown format surely 😃

Comment (netme) [reply cli/zally/utils/markdown_formatter.go]: hehe, good catch!

**Comment by netme** [comment]
@maxim-tschumak , @roxspring it's ready for review now

File: cli/zally/commands/lint.go [hunk 2]
```diff
@@ -143,3 +156,18 @@ func getReader(path string, contents []byte) readers.SpecsReader {
 	}
 	return readers.NewJSONReader(contents)
 }
+
+func getFormatter(format string) (utils.Formatter, error) {
+
+	if format == "markdown" {
```
Comment (maxim-tschumak) [inline cli/zally/commands/lint.go]: We can also use [switch case](https://tour.golang.org/flowcontrol/9).

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by netme** [comment]
👍

**Pull request merged**